core/crypto/eddsa/blinded25519.go

```diff
@@ -41,20 +32,98 @@ package eddsa
 import (
 	"crypto/ed25519"
 	"crypto/sha512"
-
+	"crypto/subtle"
+	"encoding"
+	"errors"
 	"filippo.io/edwards25519"
 )
 
+var identity_element = [32]byte{1, 0}
+
 const (
 	// BlindFactorSize is the size in bytes of the blinding factors.
 	BlindFactorSize = ed25519.PublicKeySize
 )
 
+// Sanity checking of public keys.
+// We do NOT do check for small-order points here or otherwise validate the point.
+// This function is just here to catch accidentally-bad keys, basically.
+// Checks that p != G
+// Checks that p != 1
+// Checks that p != 0
+// Checks that L*p = 1
+func CheckPublicKey(pk *PublicKey) bool {
+	var result [32]byte
+	order_64 := [64]byte{0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58, 0xd6, 0x9c,
```

Comment:
maybe put this in scope outside function body and comment as in core/crypto/edwards25519
// order is the order of Curve25519 in little-endian form.
var order = [4]uint64{0x5812631a5cf5d3ed, 0x14def9dea2f79cd6, 0, 0x1000000000000000}

Reply:
(you could also make var order Order (public) from core/crypto/edwards25519 and use that)

Reply:
I think we should delete `core/crypto/edwards25519.go`. I thought we already did. I'll make a PR for that later.